lib/loader/serialization/serialize.cpp

```diff
@@ -0,0 +1,255 @@
+#include "ast/module.h"
+#include "loader/loader.h"
+#include "serialization/serialize/serializetype.cpp"
+
+namespace WasmEdge {
+namespace Serialize {
+
+std::vector<uint8_t> Serialize::serializeModule(const AST::Module &Mod) {
+  std::vector<uint8_t> serializedModule;
+  std::vector<uint8_t> serializedCustom =
+      serializeCustomSection(Mod.getCustomSections()); // 0
+  std::vector<uint8_t> serializedType =
+      serializeTypeSection(Mod.getTypeSection()); // 1
+  std::vector<uint8_t> serializedImport =
+      serializeImportSection(Mod.getImportSection()); // 2
+  std::vector<uint8_t> serializedFunction =
+      serializeFunctionSection(Mod.getFunctionSection()); // 3
+  std::vector<uint8_t> serializedTable =
+      serializeTableSection(Mod.getTableSection()); // 4
+  std::vector<uint8_t> serializedmemery =
+      serializeMemorySection(Mod.getMemorySection()); // 5
+  std::vector<uint8_t> serializedGlobal =
+      serializeGlobalSection(Mod.getGlobalSection()); // 6
+  std::vector<uint8_t> serializedExport =
+      serializeExportSection(Mod.getExportSection()); // 7
+  std::vector<uint8_t> serializedStart =
+      serializeStartSection(Mod.getStartSection()); // 8
+  std::vector<uint8_t> serializedElement =
+      serializeElementSection(Mod.getElementSection()); // 9
+  std::vector<uint8_t> serializedCode =
+      serializeCodeSection(Mod.getCodeSection()); // 10
+  std::vector<uint8_t> serializedData =
+      serializeDataSection(Mod.getDataSection()); // 11
+  std::vector<uint8_t> serializedDataCount =
+      serializeDataCountSection(Mod.getDataCountSection()); // 12
+  return serializedModule;
+}
+
+////////////////HelperFunctions////////////////////////////////////////////////////////////////
+
+template <typename T, typename F> size_t EncodeLEB128_Impl(T d, F callback) {
+  static_assert(std::is_unsigned_v<T> && std::is_integral_v<T>);
+
+  // unsigned LEB128 encoding
+  size_t n = 0;
+  do {
+    unsigned int x = d & 0b01111111;
+    d >>= 7;
+    if (d)
+      x |= 0b10000000;
+    n++;
+    callback(x);
+  } while (d);
+
+  return n;
+}
+
+template <typename T> size_t LEB128_Len(T d) {
+  return EncodeLEB128_Impl(std::move(d), [](auto &&) {});
+}
+
+template <typename T, typename Out> Out EncodeLEB128(T d, Out out) {
+  EncodeLEB128_Impl(std::move(d), [&](uint8_t v) { *out++ = v; });
+  return out;
+}
+
+template <std::contiguous_iterator It, typename Out>
+Out EncodeRange(It f, It l, Out out, int x) {
+  using V = typename std::iterator_traits<It>::value_type;
+  static_assert(std::is_trivially_copyable_v<V>);
+
+  size_t const n = std::distance(f, l);
+  auto const *bytes = reinterpret_cast<uint8_t const *>(std::addressof(*f));
+  if (x) {
+    return std::copy(bytes, bytes + n * sizeof(V),
+                     EncodeLEB128(n, out)); // LEB128(vector.size()) vector

```

Comment:
Not to comment with the same line of code.

Reply:
Done